orttraining/orttraining/python/training/utils/data/sampler.py

```diff
@@ -0,0 +1,340 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# sampler.py
+
+import torch
+import math
+import torch.distributed as dist
+from torch.utils.data.sampler import Sampler
+from torch.utils.data.dataset import Dataset
+from typing import Optional, Iterator, Callable
+from collections import OrderedDict
+
+
+# Implementation is heavily derived from bagua/load_balancing_data_loader.py
+# https://github.com/BaguaSys/bagua/blob/01874a7c3f90904c37c5612a9db866b5d4b8b5ed/bagua/torch_api/contrib/load_balancing_data_loader.py#L12
+class LoadBalancingDistributedSampler:
+    r"""Sampler that balances the data load across workers based on the sample's complexity.
+    This sampler uses a :attr:`complexity_fn` to calculate each sample's computational
+    complexity and make each batch get similar computational complexity.
+    This is useful in scenarios like speech and NLP, where each batch has variable
```

Comment:
an example of complexity function for NLP or speech would help (sequence length?)